### ISSUE TITLE
Make JsonElementSerializer public

### ISSUE THREAD

**Issue by q-litzler** [Kotlin/kotlinx.serialization#335]
The newly added classes for serializing `JsonElement`, `JsonArray`, `JsonObject`  etc, should be public and not internal.

This would be very useful for example when using Ktor:

```kotlin
KotlinxSerializer().apply {
    it.register(JsonObjectSerializer)
}

ktor.get<JsonObject>(path)
```
#334

**Comment by irgaly** [comment]
I want this too 👍 
It's very useful parsing json to `JsonObject` .

**Comment by q-litzler** [comment]
Merged with #334
